src/relentless/optimize/method.py

```diff
@@ -335,13 +337,20 @@ def optimize(self, objective, design_variables, directory=None):
         directory : str or :class:`~relentless.data.Directory`
             Directory for writing output during optimization. Default of `None`
             requests no output is written.
+        overwrite : bool
```

Comment:
You should add `overwrite` to the base class `optimize()` method as well. I think it would also then be a good idea to move the logic for setting up the directory into a private method of `_Optimizer` so it can be reused if we implement other classes in future. Something like:
```
def _setup_directory(self, directory, overwrite):
    ...
```
doing what's in lines 361-372.